app/services/usage_monitoring/create_alert_service.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module UsageMonitoring
+  class CreateAlertService < BaseService
+    Result = BaseResult[:alert]
+
+    def initialize(organization:, subscription:, params:, billable_metric: nil)
+      @organization = organization
+      @subscription = subscription
+      @params = params
+      @billable_metric = billable_metric
+      super
+    end
+
+    def call
+      [:code, :alert_type, :thresholds].each do |field|
+        if params[field].blank?
+          return result.single_validation_failure!(field:, error_code: "#{field}_must_be_present")
```

Comment:
any reason to validate this in the service rather than adding validation on the model?

With the current approach if the 3 fields are missing you will have an error only for `code`